### PR TITLE
k8s: ingress links

### PR DESCRIPTION
**Pull request by hunchback**
WIP: no need to review at this phase

to view you can run:

```
kubectl create -f tests/k8s/ingress1.yaml
```

### CODE REVIEW

**Comment by hunchback** [comment]
run tests

**Comment by hunchback** [comment]
kindly review/merge, thanks

File: topology/probes/k8s/graph.go [hunk 1]
```diff
@@ -41,6 +41,10 @@ const (
 	nodeNameField = detailsField + ".Spec.NodeName"
 )
 
+func lookupChildren(g *graph.Graph, parent, childFilter *graph.Node, edgeFilter graph.Metadata) []*graph.Node {
```
Comment (nplanel) [inline topology/probes/k8s/graph.go]: what's the sens of this "C binding" function ?

File: topology/probes/k8s/ingress.go [hunk 2]
```diff
@@ -89,12 +148,41 @@ func (p *ingressProbe) OnDelete(obj interface{}) {
 	}
 }
 
+func (p *ingressProbe) onNodeUpdated(srvNode *graph.Node) {
```
Comment (nplanel) [inline topology/probes/k8s/ingress.go]: Shall we not need to remove the link to service on OnNodeDeleted() event ?

File: topology/probes/k8s/graph.go (hunk 1 above)
Comment (hunchback) [reply topology/probes/k8s/graph.go]: removed function (was not used in this PR context)

File: topology/probes/k8s/ingress.go (hunk 2 above)
Comment (hunchback) [reply topology/probes/k8s/ingress.go]: delNode func within `topology/graph/graph.go` already automatically deletes all edges

**Comment by hunchback** [comment]
run skydive-cdd-overview-tests

File: topology/probes/k8s/ingress.go [hunk 3]
```diff
@@ -25,24 +25,50 @@ package k8s
...
+	filter := filters.NewAndFilter(
+		filters.NewTermStringFilter("Manager", managerValue),
+		filters.NewTermStringFilter("Type", "service"),
+		filters.NewNotNullFilter("Namespace"),
+	)
+	m := graph.NewGraphElementFilter(filter)
+	return graph.NewMetadataIndexer(g, m, "Namespace")
+}
+
+func newServiceIndexerByName(g *graph.Graph) *graph.MetadataIndexer {
```
Comment (nplanel) [inline topology/probes/k8s/ingress.go]: serviceIndexerByName instance is never used

Comment (hunchback) [reply topology/probes/k8s/ingress.go]: removed

**Pull request merged**